src/qibolab/platform.py

```diff
@@ -241,8 +241,8 @@ def execute_pulse_sequences(
 
         results = defaultdict(list)
         bounds = kwargs.get("bounds", self._controller.BOUNDS)
-        for batch in batch(sequences, bounds):
-            sequence, readouts = unroll_sequences(batch, options.relaxation_time)
+        for b in batch(sequences, bounds):
+            sequence, readouts = unroll_sequences(b, options.relaxation_time)
```

Comment:
This is a more an issue than the changes in unrolling. Sorry if I haven't notice before.

Could you try with `total_ordering` and the rest, and just keeping this?

Reply:
Yeah that is what I did, this made test fail for dummy so I fixed those then I noticed the Zurich one wasn't asserting the correct sequence lenghts.

Reply:
Yeah that is what I did, this made tests fail for dummy so I fixed those then I noticed the Zurich one wasn't asserting the correct sequence lenghts.